### DISCUSSION TITLE
[A/L] (2021/22) Combinatoria - Esercizio 5

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#181] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152252336-2d59c6f3-c70c-4c90-9704-15cb7144f2fc.png)

**Comment by Exyss** [comment]
Risolvo il problema con un passaggio al complemento:
$Tot - NessunoInfo - NessunoFisica - NessunoMate = \binom{100}{5} -\binom{40}{5} -\binom{90}{5} - \binom{70}{5}$


**CORREZIONE**: la soluzione riportata sopra è errata, quella corretta è riportata in commenti successivi (si consiglia di leggere l'intera discussione al fine di evitare errori banali)

------------------------

Dopo averne discusso con @matypist, per curiosità ho controllato anche il calcolo per tipizzazione e la soluzione per tipizzazione qui sotto NON coincide con quella per complemento riportata sopra

La soluzione per tipizzazione (dunque considerando tutti i casi possibili) equivale a:

$\binom{30}{1} \binom{60}{1} \binom{10}{3} + \binom{30}{1} \binom{60}{3} \binom{10}{1} + \binom{30}{3} \binom{60}{1} \binom{10}{1} + \binom{30}{1} \binom{60}{2} \binom{10}{2} + \binom{30}{2} \binom{60}{2} \binom{10}{1} + \binom{30}{2} \binom{60}{1} \binom{10}{2} = 24.181.500$

Mentre quella per complemento corrisponde a:
$\binom{100}{5} -\binom{40}{5} -\binom{90}{5} - \binom{70}{5} = 18.577.230$

Le ipotesi sono quindi due:
- Nel passaggio al complemento viene escluso qualcosa che andrebbe riaggiunto, dunque andrebbe "aggiustato il tiro" usando il PIE e non un complemento 
- Nella tipizzazione viene contato qualche caso di troppo

**Comment by matypist** [comment]
@Exyss giusto per esplicitarli, i tipi che hai usato sono (correggimi se sbaglio):
1. M I F F F
2. M I I I F
3. M M M I F
4. M I I F F
5. M M I I F
6. M M I F F

Dove:
- Per M si intende uno studente di Matematica (30 scegli n)
- Per I si intende uno studente di Informatica    (60 scegli n)
- Per F si intende uno studente di Fisica           (10 scegli n)

Es:
1. M I F F F essendoci 1 studente di Matematica, 1 di Informatica e 3 di Fisica: (30 scegli 1) (60 scegli 1) (10 scegli 3)

**Comment by AxnNxs** [comment]
> 

La mia ipotesi è che segua il principio dell'intersezione insiemistica:

AuBuC = A + B + C - AnB - AnC - BnC + AnBnC

Quindi bisogna raggiungere probabilmente il caso in cui ci sia una delegazione che contenga almeno un elemento in ogni insieme più i rimanenti, fattore invece conteggiato nel procedimento per tipizzazione. In altre parole, il risultato giusto è il seguente:

(100 scelgo 5) - (40 scelgo 5) - (90 scelgo 5) - (70 scelgo 5) + ((30 scelgo 5) + (10 scelgo 5) + (60 scelgo 5)) = 24.181.500

**Comment by Exyss** [comment]
Giustamente i casi in cui "NessunoX" contengono anche i casi in cui "NessunoX_e_Y" ed anche "NessunoX_e_Y_e_Z", quindi bisogna effettuare un PIE standard
